### ISSUE TITLE
[Navigation Menu] Keyboard navigation gets into broken state with focus handlers

### ISSUE THREAD

**Issue by mririgoyen** [mui/base-ui#4186]
# Bug report

## Current behavior

The Navigation Menu's component, and demo, assume that a user will never want the top-level menu items to also act as links. By making this assumption, a user navigating the menu with a keyboard can open the menu by putting a menu item into focus and then activating it. However, when the top-level menu item also acts as a link, activating it will navigate the user to the `href` (it will also open the menu, but the navigation away breaks this flow).

Understandably, there is legitimate use cases for both flows. The flow that I need to implement requires that the top-level menu items are links, but the navigation menu is usable via keyboard navigation. To that end, I simply took control of the component's state via `onValueChange/value` and added `onFocus` handlers to the `Trigger`s. This allows someone using a keyboard to tab to a top-level menu item, have the menu open, and then continue to tab down through the submenu's options, ultimately ending up on the next top-level item.

However, something get's out of sync with the tab indices as a user tabs through the component in this way. If they tab all the way through the menu, and then `SHIFT+tab` to go back through, focus ends up in places I would not expect. I'll jump back to a top-level item, then to the first child of the previous top-level item's submenu, then up to it's parent. Then going forward with tabs again skips the top-level items completely and just tabs through the menus because focus is trapped on the first item.

This behavior is not accessible at all and has been very frustrating to try and overcome. Due to the nature of Base UI acting as primitives, I should be able to control the behavior in a more predictable way.

## Expected behavior

I should be able to tab and SHIFT+tab completely through a Navigation Menu in the order of available tabbable items.

## Reproducible example

StackBlitz example: https://stackblitz.com/edit/ztds4r7y?file=src%2FApp.tsx

1. Put focus into the preview and press `tab` to highligh "Overview."
2. Press `tab` until GitHub is highlighed.
3. Press `SHIFT+tab`.
   - Sometimes "Handbook" is focused -- I would expect this to be true.
   - Sometimes "Styling" within the "Handbook" menu is focused.
4. Press `SHIFT+tab`.
   - "Quick Start" within the "Overview" menu is focused -- I would expect "Overview" to be focused.
6. Keep pressing `SHIFT+tab` until "Overview" is focused again.
7. Now start `tab`ing through again until you tab through all submenu items for "Overview."
8. Notice that "Styling" is focused when you would expect "Handbook" to be focused.

## Base UI version

1.2.0

## Which browser are you using?

All

## Which OS are you using?

Mac OS

## Which assistive tech are you using (if applicable)?

Keyboard

## Additional context

Radix's Navigation Menu allows this behavior to work exactly as intended. Reluctantly, I swapped Base UI's NavigationMenu for Radix's because the functionality is the most important part of the component. Because Radix is in maintenance mode, and because Base UI's composition of the component makes it easier to animate and polish, and because I am using Base UI's primitive for nearly every other component in my library, I would prefer to swap back to Base UI once the tab issues are resolved.

**Comment by atomiks** [comment]
Have you considered doing what [Apple's navigation menu](https://www.apple.com/) does instead: 

<img width="199" height="71" alt="Image" src="https://github.com/user-attachments/assets/11a140d1-3512-4cca-8aee-cb296a42e93d" />

It shows a chevron trigger when using the keyboard but otherwise for pointer usage it remains hidden. This is a more accessible pattern than trying to hack on `onFocus` handlers. The `onFocus` technique to support a top-level link that's also a trigger has some issues:

1. A link that also triggers a submenu when it's focused mixes interaction models (navigate vs disclose)
2. On-focus opening is disorienting: tabbing across the navbar pops menus open just by landing on items, which is often discouraged (WCAG 3.2.1 "On Focus") especially for big popups like navigation menus
3. Screen reader browse vs focus mismatch: many SR users move a virtual cursor without real focus, so open on focus doesn't always fire  and when it does, it can cause noisy/odd announcements
4. Adding `aria-haspopup/expanded` to an `<a>` announces as “link” so “link, expanded” is confusing and implies conflicting primary actions
5. Focus-management edge cases (what you're hitting here): since the navigation menu popup is in a portal it uses focus guards to properly manage focus. In this case it's breaking due to the custom logic

**Comment by mririgoyen** [comment]
Thank you for your response. The example you shared from Apple is intriguing, as well as your reasonings for why my proposed pattern might be more burdensome than helpful.

I would be interesting in learning how reimplementing Apple's strategy would work with the primitives from Base UI's NavigationMenu. It would seem that the link would still need to be wrapped fully in the `Trigger`, unless it wasn't and `onMouseEnter/onMouseLeave` handlers fired to managed the state. Additonally, just the arrow couldn't be a `Trigger` if you want the menu `Positioner` to `align="start"` with the entire item.

I will mess around with this some this evening, but I think it would be excellent if this could be documented within Base UI's NavigationMenu. Additionally, an example of this kind of approach and even updates to the component itself to make implementing it more straightforward would be greatly appreciated.

If I can get something working that follows the proposed pattern, I will update here.

**Comment by atomiks** [comment]
> It would seem that the link would still need to be wrapped fully in the Trigger, unless it wasn't and onMouseEnter/onMouseLeave handlers fired to managed the state. Additonally, just the arrow couldn't be a Trigger if you want the menu Positioner to align="start" with the entire item.

Yeah, I haven't tried to do what Apple does with the Base UI navigation menu, but it seems like something we'd need to add new parts for to properly support without hacks on top (e.g. `NavigationMenu.TriggerLink`). I also tried to do this externally and it's not easy to add

**Comment by atomiks** [comment]
Here's a [workaround for v1.3.0](https://stackblitz.com/edit/xshwf691) for the top-level link that opens a menu pattern (Apple-style), which requires splitting the trigger props to individual elements

**Comment by mririgoyen** [comment]
Thank you so much for your example. I was able to use portions of it to make something work in a much more accessible fashion while maintaining our design pattern. Your work on the library, as well as your example, are much appreciated!

**Comment by mririgoyen** [comment]
@atomiks With the example pattern you provided, I'm seeing the component get into weird states where `pointer-events: none` gets applied to the `body` tag and doesn't clear. I know your example is not an officially supported method of using the component, but I wanted to reach out to see if there was any known way to disable `pointer-events: none` from being applied to the `body` tag when a sub-menu is opened. I was looking for something like the `modal` prop used on Menus, or something similar, but I haven't found a lead that works.

Just to give some more context, it seems like there is a slight race condition when a menu is open/closing, and a route change is happening via Next router in our case. The soft-nav causes the `pointer-events` to stay on the `body` and not get cleared when a nav happens to happen at the exactly right time. It's very finicky.

**Comment by atomiks** [comment]
I'll have to debug why that happens with the pattern

For now you should be able to set `pointer-events: initial; !important` on `<body>` in your CSS? Regular Menu submenus should no longer need it placed there

**Comment by mririgoyen** [comment]
Thank you. As to not introduce side-effects in other aspects of the site, doing a global `!important` on the `body` is not really something we want to do. In the meantime, I've added the following two methods to the `NavigationMenu.Root`:

```
    onOpenChangeComplete: () => {
      document.body.style.pointerEvents = '';
    }
    onValueChange: () => {
      document.body.style.pointerEvents = '';
    }
```

We can still get the issue to happen, but with these guards in place, we can at least put the mouse cursor back into the submenu and move it out to "unblock" the rest of the site. A more permanent solution would be ideal and any other suggestions you have or might come up with in the future would be much appreciated. Thank you again.
